src/component/matrix/MatrixView.ts

```diff
@@ -0,0 +1,172 @@
+
+        const xLeft = rect.x + cellWidth * yHeight;
+        for (let i = 0; i < xCells.length; i++) {
+            const cell = xCells[i];
+            const width = cellWidth * cell.colSpan;
+            const height = cellHeight * cell.rowSpan;
+            const left = xLeft + cellWidth * cell.colId;
+            const top = rect.y + cellHeight * cell.rowId;
+
+            this.group.add(new graphic.Rect({
+                shape: {
+                    x: left,
+                    y: top,
+                    width: width,
+                    height: height
+                },
+                style: xItemStyle
+            }));
+            if (xLabelModel.get('show')) {
+                this.group.add(new graphic.Text({
+                    style: createTextStyle(xLabelModel, {
+                        text: cell.value,
+                        x: left + width / 2,
+                        y: top + height / 2,
+                        verticalAlign: 'middle',
+                        align: 'center'
+                    })
+                }));
+            }
+        }
+
+        const yTop = rect.y + cellHeight * xHeight;
+        for (let i = 0; i < yCells.length; i++) {
+            const cell = yCells[i];
+            const width = cellWidth * cell.rowSpan;
+            const height = cellHeight * cell.colSpan;
+            const left = rect.x + cellWidth * cell.rowId;
+            const top = yTop + cellHeight * cell.colId;
+
+            this.group.add(new graphic.Rect({
+                shape: {
+                    x: left,
+                    y: top,
+                    width: width,
+                    height: height
+                },
+                style: yItemStyle
+            }));
+            if (yLabelModel.get('show')) {
+                this.group.add(new graphic.Text({
+                    style: createTextStyle(yLabelModel, {
+                        text: cell.value,
+                        x: left + width / 2,
+                        y: top + height / 2,
+                        verticalAlign: 'middle',
+                        align: 'center'
+                    })
+                }));
+            }
+        }
+
+        // Inner cells
+        const innerBackgroundStyle = matrixModel
+            .getModel('innerBackgroundStyle')
+            .getItemStyle();
+        for (let i = 0; i < xLeavesCnt; i++) {
+            for (let j = 0; j < yLeavesCnt; j++) {
+                const left = xLeft + cellWidth * i;
+                const top = yTop + cellHeight * j;
+                this.group.add(new graphic.Rect({
+                    shape: {
+                        x: left,
+                        y: top,
+                        width: cellWidth,
+                        height: cellHeight
+                    },
+                    style: innerBackgroundStyle
+                }));
+            }
+        }
+
+        // Outer border
+        const backgroundStyle = matrixModel
```

Comment:
These borders may overlap and cause a darker border color. I'm not sure if we should provide a similar ability like the CSS table property [`border-collapse`](https://developer.mozilla.org/docs/Web/CSS/border-collapse). It looks not easy to implement that feature.

And the chart seems rendering unexpectedly if I set the width of both the outer border and inner border to `0`.

```js
backgroundStyle: {
    borderWidth: 0
},
innerBackgroundStyle: {
    borderWidth: 0
}
```

<img src="https://github.com/apache/echarts/assets/26999792/d181125b-9c12-4365-9273-e0d115445051" width="400">

Reply:
The border won't be darker unless the opacity of borderColors are not 1. I personally think providing a `border-collapse` like mechanism is over-kill because then, we should also implement other features like border widths and colors being different values at four sides. This should bring much work to implement it robustly. The reason why I choose to render rectangles with background and border is that it would be more difficult to think which option controls which border line.

The image you provided is as expected. You should add `x.itemStyle.borderWidth: 0` and `y.itemStyle.borderWidth: 0` if you want to remove those rest borders.